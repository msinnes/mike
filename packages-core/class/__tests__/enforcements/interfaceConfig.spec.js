const typeLoader = require('../../src/loaders/type');

const interfaceConfigEnforcement = require('../../src/enforcements/interfaceConfig');

describe('interfaceConfigEnforcement', () => {
  it('should be a function', () => {
    expect(interfaceConfigEnforcement).toBeDefined();
    expect(interfaceConfigEnforcement).toBeInstanceOf(Function);
  });

  it('should not throw an error if input not a map of types', () => {
    expect(() => {
      interfaceConfigEnforcement({
        field1: typeLoader('name1', () => {}),
      });
    }).not.toThrow();
  });

  it('should throw an error if an input is not a string', () => {
    expect(() => {
      interfaceConfigEnforcement([]);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      interfaceConfigEnforcement({
        field1: 'string',
        field2: typeLoader('name1', () => {}),
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      interfaceConfigEnforcement({
        field1: 'string',
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      interfaceConfigEnforcement();
    }).toThrowErrorMatchingSnapshot();
  });
});
