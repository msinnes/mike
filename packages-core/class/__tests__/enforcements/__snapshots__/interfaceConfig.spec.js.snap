// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interfaceConfigEnforcement should throw an error if an input is not a string 1`] = `"InternalError: interface configs must be a map of types"`;

exports[`interfaceConfigEnforcement should throw an error if an input is not a string 2`] = `"InternalError: interface configs must be a map of types"`;

exports[`interfaceConfigEnforcement should throw an error if an input is not a string 3`] = `"InternalError: interface configs must be a map of types"`;

exports[`interfaceConfigEnforcement should throw an error if an input is not a string 4`] = `"InternalError: interface configs must be a map of types"`;
